### ISSUE TITLE
Support URL field/property for events

### ISSUE THREAD

**Issue by pReya** [nextcloud/calendar#2895]
**Is your feature request related to a problem? Please describe.**
I couldn't find any existing issue for this, even though I can't imagine this has never been asked/suggested (am I just too blind to finde it?).

Apples calendar supports a "URL" property on events. Not sure if this is part of the standard or an  Apple proprietary addition. Anyway it would be great if the URL field was supported by Nextcloud calendar.

**Describe the solution you'd like**
Show the given URL in the calendar event details, make it clickable.

![Screenshot 2021-02-17 at 10 39 49](https://user-images.githubusercontent.com/4677417/108185654-d27ebe80-710c-11eb-8df3-a37d06869bd0.png)

**Comment by tcitworld** [comment]
Seems acceptable to me to add it, but it *should* only be used to convey "a more dynamic rendition of the calendar information associated with the calendar component can be found", otherwise there's also a [`CONFERENCE`](https://github.com/nextcloud/calendar/issues/1758) [standardized](https://tools.ietf.org/html/rfc7986#section-5.11) property which might be more suited for most usages (including ours with Nextcloud Talk).

Opinions @nextcloud/designers ?

**Comment by pReya** [comment]
> Seems acceptable to me to add it, but it _should_ only be used to convey "a more dynamic rendition of the calendar information associated with the calendar component can be found", otherwise there's also a [`CONFERENCE`](https://github.com/nextcloud/calendar/issues/1758) [standardized](https://tools.ietf.org/html/rfc7986#section-5.11) property which might be more suited for most usages (including ours with Nextcloud Talk).
> 
> Opinions @nextcloud/designers ?

Thanks! I did not know about the `conference` property. But still, I typically use this URL to link to public websites regarding the given event (imagine a Meetup page or a concert website etc. – so nothing related to video conferencing).

**Comment by fhaut** [comment]
![image](https://user-images.githubusercontent.com/2178618/175179032-fd71b882-1a7d-4274-b43a-763e97b77bdb.png)


I understand that in a text field evoluate a create a link is more dificult, but if there is a specific field called URL, with a button side that opens a new tab with the URL field is possible?

only that feature missing for leaving google calendar

thank's

**Comment by nimishavijay** [comment]
I think this is a good idea :) as @pReya mentioned the URL field can be used for websites related to the event. However, since there are 2 standard properties for URL and conferencing, we would have to name them distinctly, for eg. "Website" and "Meeting link". If there is only one field which accepts any URL it would likely be filled with both resources and meeting links and may cause confusion if exported to other calendar software which show this distinction.

**Comment by ZID-TU-Graz** [comment]
The customers of our self hosted instance would also like to see this feature.

**Comment by szaimen** [comment]
@st3iny @ChristophWurst how important is this now after https://github.com/nextcloud/calendar/pull/5355? How many dav clients even implement this field btw?

**Comment by st3iny** [comment]
We use the location field for talk room links so we kind of implemented it already. Also, links in the location field are now clickable (since the issue linked above).

**Comment by pReya** [comment]
> @st3iny @ChristophWurst how important is this now after #5355? How many dav clients even implement this field btw?

I think these are very separate topics.

> We use the location field for talk room links so we kind of implemented it already. Also, links in the location field are now clickable (since the issue linked above).

As stated before, I don't think this is a good idea:
- What about events that have both, a physical location AND a videocall?
- The location field is handled very differently in all clients – many don't allow links there, e.g. users will have to copy & past the link to open it.

**Comment by keunes** [comment]
Just a note that Etar (Android calendar app) just implemented the/a URL field: https://github.com/Etar-Group/Etar-Calendar/releases/tag/v1.0.35

(Not sure it it could get synchronised via DAVx.)

**Comment by tcitworld** [comment]
It is, see https://github.com/Etar-Group/Etar-Calendar/pull/1357, https://github.com/Etar-Group/Etar-Calendar/issues/903 and https://manual.davx5.com/integration.html#extended-event-properties

The issue still makes sense, just lower priority for now.

**Comment by jbeyerstedt** [comment]
Just to add to the possible use cases of the URL field: We would like to use the Nextcloud calendar as the data source of events in our local hack space. Each event shall link to a wiki page for recurring events or blog post with details. On our website we have a list of upcoming events (generated from the jcal file) and it would be nice, if this list on the website can link to the pages without parsing links from the description field.

**Comment by pReya** [comment]
> Just to add to the possible use cases of the URL field: We would like to use the Nextcloud calendar as the data source of events in our local hack space. Each event shall link to a wiki page for recurring events or blog post with details. On our website we have a list of upcoming events (generated from the jcal file) and it would be nice, if this list on the website can link to the pages without parsing links from the description field.

I've built the same functionality for my local hackspace, and we're just using the physical "location" field for the link. It's not nice, but it works.
